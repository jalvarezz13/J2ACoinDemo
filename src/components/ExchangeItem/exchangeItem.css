/* Exchange Item */
:root {
  --coinbase: #0d70e2;
  --binance: #f3ba2f;
  --crypto: #242f61;
  --bit2me: #2a4d9a;
}

.exchange-item-container {
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
  justify-content: center;

  min-width: 15%;
  height: 100%;
}

.exchange-item-logo {
  height: 200px;
  padding-bottom: 10px;
  -webkit-filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));
  filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));
}

.exchange-rrss-container {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: center;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
}

.exchange-rrss-item {
  width: 25px;
}
.exchange-rrss-item:hover {
  transform: translateY(-2px);
  transition: 0.2s;
}

.exchange-visit-container {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: center;
  justify-content: space-evenly;
  align-items: center;
  padding-bottom: 10px;
}

.exchange-visit-item {
  width: 25px;
}

.exchange-progressBar {
  width: 200px;
  padding-bottom: 10px;
}

.progress-bar {
  width: 100%;
  background-color: #e0e0e0;
  padding: 2px;
  border-radius: 5px;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.2);
}

.progress-bar-fill {
  display: block;
  height: 22px;
  border-radius: 5px;
  font-weight: 700;
}

.coinbase-exchange {
  background-color: var(--coinbase);
}

.binance {
  background-color: var(--binance);
}

.crypto-com-exchange {
  background-color: var(--crypto);
}

.bit2me {
  background-color: var(--bit2me);
}

.progress-bar-loading {
  width: 25px;
}

.progress-bar-icon1 {
  width: 15px;
  margin: 0 5px;
}

.progress-bar-icon2 {
  width: 13px;
  margin: 0 5px;
}

.progress-bar-fill-container {
  display: flex;
  flex-direction: row;
  align-content: center;
  justify-content: center;
  align-items: center;
}
