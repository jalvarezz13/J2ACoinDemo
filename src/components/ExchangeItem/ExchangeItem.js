import React from "react"
import PropTypes from "prop-types"

import "./exchangeItem.css"

import blogIcon from "../../assets/blog.svg"
import feeIcon from "../../assets/fee.svg"
import telegramIcon from "../../assets/telegram.svg"
import twitterIcon from "../../assets/twitter.svg"
import webIcon from "../../assets/web.svg"
import chartIcon from "../../assets/chart.svg"
import shieldIcon from "../../assets/shield.svg"
import lockIcon from "../../assets/lock.svg"

const ExchangeItem = ({ item }) => {
  const rrssLinks = Object.values(item.urls)

  return (
    <div className="exchange-item-container">
      <img className="exchange-item-logo" src={require(`../../assets/${item.id}.svg`)} alt="exchange logo"></img>
      <h2 style={{ paddingBottom: "5px" }}>{item.name}</h2>

      <div className="exchange-progressBar">
        <div className="progress-bar">
          <span className={`progress-bar-fill ${item.slug}`} style={{ width: `${item.security}%` }}>
            <div className="progress-bar-fill-container">
              <img className="progress-bar-icon1" src={shieldIcon} alt="" />
              {`${item.security}%`}
              <img className="progress-bar-icon2" src={lockIcon} alt="" />
            </div>
          </span>
        </div>
      </div>

      <div className="exchange-visit-container">
        <img className="exchange-visit-item" src={chartIcon} alt="visits icon" />
        <span style={{ width: "15px" }}></span>
        <p>{new Intl.NumberFormat("de-DE").format(item.weekly_visits)} visitas</p>
      </div>

      <div className="exchange-rrss-container">
        {rrssLinks[4].length !== 0 && (
          <a href={rrssLinks[4][0]} title="Exchange" target="_blank" rel="noreferrer">
            <img className="exchange-rrss-item" src={webIcon} alt="web icon" />
          </a>
        )}
        {rrssLinks[0].length !== 0 && (
          <a href={rrssLinks[0][0]} title="Blog" target="_blank" rel="noreferrer">
            <img className="exchange-rrss-item" src={blogIcon} alt="blog icon" />
          </a>
        )}
        {rrssLinks[2].length !== 0 && (
          <a href={rrssLinks[2][0]} title="Tarifas" target="_blank" rel="noreferrer">
            <img className="exchange-rrss-item" src={feeIcon} alt="fee icon" />
          </a>
        )}
        {rrssLinks[3].length !== 0 && (
          <a href={rrssLinks[3][0]} title="Twitter" target="_blank" rel="noreferrer">
            <img className="exchange-rrss-item" src={twitterIcon} alt="twitter icon" />
          </a>
        )}
        {rrssLinks[1].length !== 0 && (
          <a href={rrssLinks[1][0]} title="Telegram" target="_blank" rel="noreferrer">
            <img className="exchange-rrss-item" src={telegramIcon} alt="telegram icon" />
          </a>
        )}
      </div>
    </div>
  )
}

ExchangeItem.propTypes = {
  item: PropTypes.object.isRequired,
}

export default ExchangeItem
