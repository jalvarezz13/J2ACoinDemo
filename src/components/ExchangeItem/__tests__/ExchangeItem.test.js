import React from "react"
import { shallow } from "enzyme"
import ExchangeItem from "../ExchangeItem"

const rightJson = {
  countries: [],
  date_launched: "2014-07-22T00:00:00.000Z",
  description:
    "Bit2Me is an international ecosystem of crypto solutions that has been growing since 2014, being a leader in Southern Europe with the main mission to generate cryptocurrency education and adoption.\r\nOur Education division boasts of being virtually the largest crypto academy (articles, udemy-format courses with certificates, audio/visual format etc.) in the Spanish speaking world.\r\nWith this, we have 20 different crypto adoption tools and services wherein the B2M token is at the center of everything we do, including our academy. We currently count on 450K verified Bit2Me Wallet users and more than 3 million alumni in Bit2Me Academy from 100 different nationalities.  \r\n\r\nOur competitors also offer a full crypto digital suite. However, none of them focus on our niche demographic wherein we provide customer care, telephone support, and customer success account managers for small accounts in native Spanish, Portuguese, and Italian. Our main focus has always been Southern Europe and Latin America. In fact, half of\r\nour 3 million alumni in our Bit2Me Academy this year are from LATAM, and 80% of our 450K verfified users are residents of iberia peninsula and Italy.",
  fiats: [],
  id: 1561,
  is_hidden: 0,
  is_redistributable: null,
  logo: "https://s2.coinmarketcap.com/static/img/exchanges/64x64/1561.png",
  maker_fee: 0,
  name: "Bit2Me",
  notice: "",
  slug: "bit2me",
  spot_volume_last_updated: "2022-03-27T16:45:15.747Z",
  spot_volume_usd: 14115505.99102702,
  tags: null,
  taker_fee: 0,
  type: "",
  urls: {
    blog: ["https://medium.com/@bit2me"],
    chat: ["https://t.me/Bit2Me_ES"],
    fee: ["https://support.bit2me.com/en/support/solutions/articles/35000172197-what-are-the-fees-for-the-trade-service-"],
    twitter: ["https://twitter.com/bit2me"],
    website: ["https://trade.bit2me.com/exchange/BTC-EUR"],
  },
  weekly_visits: 820,
}
const wrongJson = {
  countries: [],
  date_launched: "2014-07-22T00:00:00.000Z",
  description:
    "Bit2Me is an international ecosystem of crypto solutions that has been growing since 2014, being a leader in Southern Europe with the main mission to generate cryptocurrency education and adoption.\r\nOur Education division boasts of being virtually the largest crypto academy (articles, udemy-format courses with certificates, audio/visual format etc.) in the Spanish speaking world.\r\nWith this, we have 20 different crypto adoption tools and services wherein the B2M token is at the center of everything we do, including our academy. We currently count on 450K verified Bit2Me Wallet users and more than 3 million alumni in Bit2Me Academy from 100 different nationalities.  \r\n\r\nOur competitors also offer a full crypto digital suite. However, none of them focus on our niche demographic wherein we provide customer care, telephone support, and customer success account managers for small accounts in native Spanish, Portuguese, and Italian. Our main focus has always been Southern Europe and Latin America. In fact, half of\r\nour 3 million alumni in our Bit2Me Academy this year are from LATAM, and 80% of our 450K verfified users are residents of iberia peninsula and Italy.",
  fiats: [],
  id: 1561,
  is_hidden: 0,
  is_redistributable: null,
  logo: "https://s2.coinmarketcap.com/static/img/exchanges/64x64/1561.png",
  maker_fee: 0,
  name: "Bit2Me4",
  notice: "",
  slug: "bit2me",
  spot_volume_last_updated: "2022-03-27T16:45:15.747Z",
  spot_volume_usd: 14115505.99102702,
  tags: null,
  taker_fee: 0,
  type: "",
  urls: {
    blog: ["https://medium.com/@bit2me"],
    chat: ["https://t.me/Bit2Me_ES"],
    fee: ["https://support.bit2me.com/en/support/solutions/articles/35000172197-what-are-the-fees-for-the-trade-service-"],
    twitter: [],
    website: [],
  },
  weekly_visits: 820,
}

describe("CryptoRankingItem tests", () => {
  test("Should <ExchangeItem /> renders correctly", () => {
    const wrapper = shallow(<ExchangeItem item={rightJson} />)
    expect(wrapper).toMatchSnapshot()
  })

  test("Should show an incorrect name", () => {
    const wrapper2 = shallow(<ExchangeItem item={wrongJson} />)
    const nameExchange = wrapper2.find("h2").text().trim()
    expect(nameExchange).toBe("Bit2Me4")
  })

  test("Should show five rrss", () => {
    const wrapper3 = shallow(<ExchangeItem item={rightJson} />)
    const rrssCount1 = wrapper3.find("a").length
    expect(rrssCount1).toBe(5)
  })

  test("Should show four rrss", () => {
    const wrapper4 = shallow(<ExchangeItem item={wrongJson} />)
    const rrssCount2 = wrapper4.find("a").length
    expect(rrssCount2).toBe(3)
  })
})
