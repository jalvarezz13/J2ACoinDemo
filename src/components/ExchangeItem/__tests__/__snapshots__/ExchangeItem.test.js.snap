// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoRankingItem tests Should <ExchangeItem /> renders correctly 1`] = `ShallowWrapper {}`;
